src/rust/lib/system/web/src/lib.rs

```diff
@@ -503,7 +503,35 @@ pub async fn sleep(duration:Duration) {
 #[cfg(not(target_arch = "wasm32"))]
 pub use async_std::task::sleep;
 
+/// Stores arguments extracted from Location search. e.g. http://localhost/?arg0=value0&arg1=value1.
+#[derive(Shrinkwrap,Debug)]
+pub struct Arguments {
+    pub hash_map : HashMap<String,String>
+}
+
+impl Arguments {
+    fn args_from_search(search:&str) -> HashMap<String,String> {
+        if search.chars().nth(0) == Some('?') {
+            search[1..].split("&").filter_map(|arg| {
+                let x : Vec<&str> = arg.split("=").collect();
+                match x.len() {
+                    1 => Some((x[0].to_string(), "".to_string())),
+                    2 => Some((x[0].to_string(), x[1].to_string())),
+                    _ => None
```

Comment:
this code can be made safer. It uses `x.len()` to guarantee that `x[0]` or `x[1]` do not panic, but instead you could use `head` and `tail` and Options to do it the same way but safer regarding future code refactoring.